weed/filer/reader_at.go

```diff
@@ -98,10 +98,9 @@ func (c *ChunkReadAt) Close() error {
 
 func (c *ChunkReadAt) ReadAt(p []byte, offset int64) (n int, err error) {
 
-	c.readerPattern.MonitorReadAt(offset, len(p))
-
 	c.readerLock.Lock()
 	defer c.readerLock.Unlock()
+	c.readerPattern.MonitorReadAt(offset, len(p))
```

Comment:
why this is needed?

Reply:
the MonitorReadAt is not concurrent safe, I think call this func after lock is high Robustness.